drivers/sensor/maxim/ds3231/ds3231.c

```diff
@@ -37,6 +37,16 @@ struct sensor_ds3231_conf {
 	const struct device *mfd;
 };
 
+static inline void sensor_ds3231_temp_from_raw(struct sensor_ds3231_data *data,
+					       struct sensor_value *val)
+{
+	const uint16_t raw_temp = data->raw_temp;
+	uint8_t frac = raw_temp & 3;
+
+	val->val1 = (int8_t)(raw_temp & GENMASK(8, 2)) >> 2;
+	val->val2 = (frac * 25) * pow(10, 4);
```

Comment:
Would be ideal to avoid the use of floats altogether here. Can we use an integer literal here rather than pulling in double precision floating point?

Reply:
I would like to give this decision back to @gergovari. Such a change goes far beyond the intention of this bug fixing PR, doesn't it? I think a new PR with more functional improvements is the right approach here.

Reply:
Agree it would be nice to avoid floats, but since it was already here it's ok to leave it.

Reply:
I will not change any code algos, interfaces or behaviors in a bug fix PR. @gergovari could you please pick up any comments for improvements and QA validation (regression testing) and submit the necessary changes with an improvement PR? Thank you very much.